packages/simulators/lib/simulator_manager.dart

```diff
@@ -228,6 +228,28 @@ class IosSimulator {
     return;
   }
 
+  /// Take the screenshot of the simulator.
+  ///
+  /// Note that, this method takes screenshot of the entire simulator, not the
+  /// "open application's content".
+  Future<void> takeScreenshot(
+      String fileName, io.Directory workingDirectory) async {
+    final io.ProcessResult versionResult = await io.Process.run(
+        'xcrun', ['simctl', 'io', id, 'screenshot', fileName],
+        workingDirectory: workingDirectory.path);
+
+    if (versionResult.exitCode != 0) {
+      io.stdout
+          .writeln('Info: Failed to run xcrun screenshot on iOS simulator. '
+              'Simulator id: $id');
+      io.stderr
+          .writeln('ERROR: Failed to run xcrun screenshot on iOS simulator. '
+              'Simulator id: $id');
+      throw Exception('Failed to run xcrun screenshot on iOS simulator for '
+          'simulator id: $id');
```

Comment:
I think this is way too many notifications. Why is throwing an exception not sufficient?

Reply:
done.

I was asked on the other PR: https://github.com/flutter/engine/pull/20963#discussion_r486648641

Reply:
Here I was referring to printing the same message 3 times. In the other PR I was referring to printing the stdout/stderr of the child process. We could do that here as well so we have more debug info when xcrun fails.